Automattic/jetpack
# General: remove deprecated core get_page_by_title

## Proposed changes:

See #27795

Let's use direct WP queries instead, as recommended:
https://make.wordpress.org/core/2023/03/06/get_page_by_title-deprecated/

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

The changes in Jetpack's `functions.global.php` are not really used anymore. We could probably get rid of all that  logic at this point; the code exists to facilitate the migration from our own widget (https://github.com/Automattic/jetpack/blob/3104e377ae2d45ad2d4ffadfd839cee47aa871df/projects/plugins/jetpack/modules/widgets/image-widget.php#L25) to the Core widget (see #7103). This happened in WP 4.8, so we could probably remove our widget and the logic. Created #29458 to keep track of that.

The change to the Debug Helper like so;

* Activate the Jetpack Debug plugin
* Go to `/wp-admin/admin.php?page=jetpack-debug-tools`
* Activate the "Jetpack Scan Helper" module
* Go to `/wp-admin/admin.php?page=scan-helper`
* Activate each threat and validate that it is added to the site, via the Protect plugin.
* Deactivate each threat and validate that it is removed from the site.
